internal/test/factory/block.go

```diff
@@ -89,13 +85,12 @@ func MakeHeader(h *types.Header) (*types.Header, error) {
 		h.ProposerAddress = RandomAddress()
 	}
 
-	return h, h.ValidateBasic()
-}
+	require.NoError(t, h.ValidateBasic())
 
-func MakeRandomHeader() *types.Header {
-	h, err := MakeHeader(&types.Header{})
-	if err != nil {
-		panic(err)
-	}
 	return h
 }
+
+func MakeRandomHeader(t *testing.T) *types.Header {
+	t.Helper()
+	return MakeHeader(t, &types.Header{})
+}
```

Comment:
This probably deserves documentation if we're going to keep it. That said: I recommend we omit it entirely, use `MakeHeader(t, new(types.Header))` directly, and update the documentation there. The randomness is implemented there already anyway.

Say, for example:

```go
// MakeHeader fills the unset fields of h in-place with random legal values, so
// that the resulting header passes basic validation checks. It returns h.
```

or words to that effect.

Reply:
I removed it, as it's only used in one place and is kind of silly.